Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/SurfaceMagnetism.cs

```diff
@@ -155,13 +156,34 @@ private Vector3 RaycastDirection
         /// <summary>
         /// A constant scale override may be specified for volumetric raycasts, otherwise uses the current value of the solver link's alt scale
         /// </summary>
-        private float ScaleOverride => useLinkedAltScaleOverride ? SolverHandler.AltScale.magnitude : volumeCastSizeOverride;
+        private float ScaleOverride => useLinkedAltScaleOverride ? SolverHandler.AltScale.Current.magnitude : volumeCastSizeOverride;
 
-        private void OnValidate()
+        protected override void OnValidate()
         {
+            base.OnValidate();
+
             if (raycastMode == RaycastModeType.Box)
             {
-                boxCollider = gameObject.EnsureComponent<BoxCollider>();
```

Comment:
Why remove this and add error for developer to manually add one?

Reply:
A few reasons:

1. `AddComponent` (called by `EnsureComponent`) cannot be called in `OnValidate`.
2. Why would we potentially add a second, unwanted collider instead of informing the developer to add it correctly?

![image](https://user-images.githubusercontent.com/3580640/45447268-dc6e7e80-b683-11e8-8f29-9c75831e53fd.png)

Reply:
good points.